packages/components/src/components/Snackbar/Snackbar.tsx

```diff
@@ -75,10 +54,36 @@ const SnackbarButton: FC<SnackbarButtonProps> = ({ text, onPress }) => {
   )
 }
 
-export type SnackbarProps = Omit<ToastProps, 'data' | 'message'> &
-  snackbarData & {
-    message: string
-  }
+/**
+ * All options associated with the useSnackbar.show function
+ */
+export type SnackbarOptions = SnackbarData & {
+  /** offset from bottom of screen. defaults to 50 in SnackbarProvider */
+  offset?: number
+}
+
+/**
+ * Optional data passed into Snackbar component
+ */
+export type SnackbarData = {
+  /** true if snackbar represents an error state */
+  isError?: boolean
+  /** message text A11y label override */
+  messageA11y?: string
+  /** action button onPress logic for "Try again" (isError=true) or "Undo" button */
+  onActionPressed?: () => void
+}
+
+// List of ToastProps needed within Snackbar
+type activeToastProps = 'onHide'
+
+/**
+ * Properties used by Snackbar component
+ */
+export type SnackbarProps = Pick<ToastProps, activeToastProps> & {
+  data?: SnackbarData
+  message: string
+}
```

Comment:
This is the part mentioned in standup that I'm not sure is worth it.

On the one hand, this nicely delineates the typing into purposeful bins:
- Options - what end users can send to `useSnackbar.show`
- Data - what goes into the actually rendered Snackbar (excluding toast-level thing like offset, potentially duration)
- Props - what's needed in the rendered Snackbar

On the other hand, it adds complexity that I'm not sure is worth it vs:
- Merge SnackbarData into SnackbarOptions and:
```
export type SnackbarProps = Omit<ToastProps, 'data' | 'message'> &
  {
    data?: SnackbarOptions
    message: string
  }
```
and having a bunch of extra irrelevant stuff (both library-level and our props that are irrelevant in the component itself) in the Snackbar when using `toast.`

Reply:
I'm in favor of the solution that locks it down more which sounds like its the more complex one. 

Is there a functional reason why `offset` isn't just included in `SnackbarData`, or is it solely to separate them by purpose? That could be a middle ground that is still relatively clean.

Reply:
`offset` is separate because it's not relevant to pass into the Snackbar itself since it's only relevant to the hook/SnackbarProvider. So in the Snackbar you'd be able to reference `toast.offset` which is meaningless by that point.

In terms of what the consumer sees, this typing is no different than that simplified example. 

It is purely for our benefit and a fairly minor one at that--basically just controlling what appears in Snackbar when you do `toast.`. So really the benefit question is about if having it structured helps make it clearer what each piece is for or if we don't really care that a bunch of needless props appear for `toast.` and want to keep it simple.

Reply:
Okay, I understand now. Thinking long term for engineers that join or if we circle back, if we're pretty sure those additional props from `toast.` aren't needed, I think it could be useful to hide them so they don't show up in Intellisense.